src/app/features/students/students.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, OnDestroy, OnInit, Signal, WritableSignal, computed, effect, inject, signal } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+
+import { NzTypographyModule } from 'ng-zorro-antd/typography';
+import { NzTableModule } from 'ng-zorro-antd/table';
+import { NzDividerModule } from 'ng-zorro-antd/divider';
+import { NzButtonModule } from 'ng-zorro-antd/button';
+import { NzInputModule } from 'ng-zorro-antd/input';
+import { NzPopconfirmModule } from 'ng-zorro-antd/popconfirm';
+
+import { StudentService } from '../../shared/services/student.service';
+import { Student, StudentModalType } from '../../shared/interfaces/students';
+import { HeaderComponent } from '../../shared/components/header/header.component';
+import { NzMessageModule, NzMessageService } from 'ng-zorro-antd/message';
+import { StudentModalComponent } from './components/student-modal/student-modal.component';
+import { Subject, takeUntil } from 'rxjs';
+
+@Component({
+  selector: 'app-students',
+  standalone: true,
+  imports: [CommonModule, HeaderComponent, NzTypographyModule, NzTableModule, NzDividerModule, NzButtonModule, NzInputModule, FormsModule, CommonModule, NzMessageModule, NzPopconfirmModule, StudentModalComponent],
+  templateUrl: './students.component.html',
+  styleUrl: './students.component.sass'
+})
+export class StudentsComponent implements OnInit, OnDestroy {
+  private studentsService: StudentService = inject(StudentService)
+  private toast: NzMessageService = inject(NzMessageService)
+  
+  public isShowStudentModal: boolean = false;
+
+  public students: Signal<Student[]> = computed(() => this.studentsService.students())
+  public loading: Signal<boolean> = computed(() => this.studentsService.loading())
+  private error: Signal<string | null> = computed(() => this.studentsService.error())
+  public actionModal: WritableSignal<StudentModalType> = signal("add");
+
+  private onDestroy$: Subject<void> = new Subject();
+
+  constructor() {
+    effect(() => {
+      if (this.error()) {
+        this.toast.error(this.error() || '', {
+          nzDuration: 5000
+        })
+      }
+    })
+  }
+
+  public ngOnInit(): void {
+    this.studentsService.getStudents().pipe(takeUntil(this.onDestroy$)).subscribe();
+    this.studentsService.deleteStudent().pipe(takeUntil(this.onDestroy$)).subscribe();
+    this.studentsService.selectStudent().pipe(takeUntil(this.onDestroy$)).subscribe();
+  }
+  
+  public showStudentModal({ action, student }: { action: StudentModalType, student: null | Student }): void {
+    this.isShowStudentModal = true;
+    this.actionModal.set(action);
+    if (student) {
+      this.studentsService.selectedStudentSubject$.next(student);
+    }
+  }
+
+  public deleteStudent(id: string): void {
+    this.studentsService.deleteStudentSubject$.next(id);
```

Comment:
Instead of pushing student that you want to delete, you should call subscribe here. Method deleteStudent() in your service should know already what student must be deleted. Subject value should be filled first, then you call delete method